ddclient

```diff
@@ -2298,57 +2281,72 @@ sub get_ip {
 }
 
 ######################################################################
-## is_ipv4() validates if string is valid IPv4 address and only
-## a valid IPv4 address (no preceding or trailing spaces/characters)
+## is_ipv4() validates if string is valid IPv4 address and only a
+## valid address with no preceding or trailing spaces/characters
+## and no embedded leading zeros.
 ######################################################################
 sub is_ipv4 {
     my ($value) = @_;
     return (length($value // '') != 0) && ($value eq (ipv4_match($value) // ''));
 }
 
 ######################################################################
-## ipv4_match() extracts the first valid IPv4 address from given string
+## ipv4_match() extracts the first valid IPv4 address from given string.
+## Accepts leading zeros in the address but removes them in returned value
 ######################################################################
 sub ipv4_match {
-    (shift // '') =~ /\b((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))\b/ai;
-    return $1;
+    (shift // '') =~ /\b((?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3} ## first three bytes
+                         (?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))\b/xai;  ## last one
+    return ($1 // '') =~ s/\b0+\B//gr;                                     ## remove leading zeros
```

Comment:
The `/r` flag is only available in Perl 5.14+ so it can't be used. (We support Perl 5.10.1 and later.)

Reply:
Well that's a bummer.  And I was so proud of myself as its rather elegant.  Oh well, have to create a local variable.